### PR TITLE
Linux work

### PR DESCRIPTION
**Pull request by jpcima**
Hi @lundstroem
(ping, as I noticed you're not member of the repository watch list)

This PR is the current Linux work from linuxmao-org/master. (+ misc minor things)

#1

### CODE REVIEW

File: snibbetracker/src/main.c [hunk 1]
```diff
@@ -4588,9 +4594,13 @@ static int get_buffer_size_from_index(int i) {
 
 }
 
-static void copy_project_win(const char *name) {
+static void copy_project_win_lin(const char *name) {
     char *read_path = cAllocatorAlloc((1024 * sizeof(char*)), "win path 1");
+    #if defined(platform_windows)
     snprintf(read_path, 1023, "%s%s", "demos\\", name);
+    #else
+    snprintf(read_path, 1023, "%s/../share/lundstroem/snibbetracker/demos/%s", base_dir, name);
```
Comment (lundstroem) [inline snibbetracker/src/main.c:4602]: is base_dir NULL here?

Comment (jpcima) [reply snibbetracker/src/main.c:4602]: I obtain the base directory from SDL in the main function.
If this fails with NULL for any reason, I exit with the code 1.
Therefore, NULL is impossible.

Comment (lundstroem) [reply snibbetracker/src/main.c:4602]: Ah that is correct, was a little difficult following the execution order in the diff :)

**Review by lundstroem — APPROVED** [review]

**Comment by trebmuh** [comment]
:+1:

**Pull request merged**